components/camera/videosource/webcam.go

```diff
@@ -217,48 +216,125 @@ func makeConstraints(conf *WebcamConfig, debug bool, logger logging.Logger) medi
 	}
 }
 
+// getNamedVideoSource is a helper function for trying to open a webcam that attempts to
+// find a video device (not a screen) by the given name.
+//
+// First it will try to use the path name after evaluating any symbolic links. If
+// evaluation fails, it will try to use the path name as provided.
+func getNamedVideoSource(
+	path string,
+	fromLabel bool,
+	constraints mediadevices.MediaStreamConstraints,
+	logger logging.Logger,
+) (gostream.MediaSource[image.Image], error) {
+	if !fromLabel {
+		resolvedPath, err := filepath.EvalSymlinks(path)
+		if err == nil {
+			path = resolvedPath
+		}
+	}
+	return gostream.GetNamedVideoSource(filepath.Base(path), constraints, logger)
+}
+
+// tryWebcamOpen is a helper for finding and making the video source that uses getNamedVideoSource to try and find
+// a video device (gostream.MediaSource). If successful, it will wrap that MediaSource in a VideoSource.
+func tryWebcamOpen(
+	ctx context.Context,
+	conf *WebcamConfig,
+	path string,
+	fromLabel bool,
+	constraints mediadevices.MediaStreamConstraints,
+	logger logging.Logger,
+) (gostream.VideoSource, error) {
+	source, err := getNamedVideoSource(path, fromLabel, constraints, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if conf.Width != 0 && conf.Height != 0 {
+		img, release, err := gostream.ReadMedia(ctx, source)
+		if release != nil {
+			defer release()
+		}
+		if err != nil {
+			return nil, err
+		}
+		if img.Bounds().Dx() != conf.Width || img.Bounds().Dy() != conf.Height {
+			return nil, errors.Errorf("requested width and height (%dx%d) are not available for this webcam"+
+				" (closest driver found by gostream supports resolution %dx%d)",
+				conf.Width, conf.Height, img.Bounds().Dx(), img.Bounds().Dy())
+		}
+	}
+	return source, nil
+}
+
+// getPathFromVideoSource is a helper function for finding and making the video source that
+// returns the path derived from the underlying driver via MediaSource or an empty string if a path is not found.
+func getPathFromVideoSource(src gostream.VideoSource, logger logging.Logger) string {
+	labels, err := gostream.LabelsFromMediaSource[image.Image, prop.Video](src)
+	if err != nil || len(labels) == 0 {
+		logger.Errorw("could not get labels from media source", "error", err)
+		return ""
+	}
+
+	return labels[0] // path is always the first element
+}
+
 // findAndMakeVideoSource finds a video device and returns a video source with that video device as the source.
 func findAndMakeVideoSource(
 	ctx context.Context,
 	conf *WebcamConfig,
-	label string,
+	path string,
 	logger logging.Logger,
 ) (gostream.VideoSource, string, error) {
 	mediadevicescamera.Initialize()
 	debug := conf.Debug
 	constraints := makeConstraints(conf, debug, logger)
-	if label != "" {
-		cam, err := tryWebcamOpen(ctx, conf, label, false, constraints, logger)
+	if path != "" {
```

Comment:
label is a term used in mediadevices to broadly describe data associated with a driver (device name, manufacturer, path etc.). We should be more specific since we get the path label specifically here

Reply:
Nice! I agree with this rename to avoid confusion with `label` in our mediadevices path. 

Only caveat here is webcams on Darwin which use a hash/label for for the avfoundation backend.

Reply:
Forgot about that. Ok with keeping path?

Reply:
I'm ok with with keeping path -- I am almost always thinking in terms of Linux.